mbs_results/staging/back_data.py

```diff
@@ -0,0 +1,161 @@
+import pandas as pd
+
+from mbs_results.utilities.utils import (
+    convert_column_to_datetime,
+    read_colon_separated_file,
+)
+
+
+def is_back_data_date_ok(
+    back_data_period: pd.Series,
+    first_period: pd.Timestamp,
+    current_period: int,
+    revision_period: int,
+):
+    """
+    Applies checks on period 0 (back data period)
+
+    Staged data need back data (or period 0) to succesfully impute values,
+    this function ensures the date in back data is correctly configured.
+
+    Parameters
+    ----------
+    back_data_period : pd.Series
+        All dates in back data.
+    first_period : pd.Datetime
```

Comment:
first period is pandas dataframe, is this correct. function def says timestamp :)

Reply:
good spot !